router/batchrouter/batchrouter.go

```diff
@@ -422,15 +422,18 @@ func (brt *HandleT) pollAsyncStatus(ctx context.Context) {
 											statusList = append(statusList, status)
 										}
 										brt.failedJobCount.Count(len(statusList))
-										txn := brt.jobsDB.BeginGlobalTransaction()
-										brt.jobsDB.AcquireUpdateJobStatusLocks()
-										err = brt.jobsDB.UpdateJobStatusInTxn(txn, statusList, []string{brt.destType}, parameterFilters)
+										err := brt.jobsDB.WithUpdateSafeTx(func(tx jobsdb.UpdateSafeTx) error {
```

Comment:
Don't get the structure of the interface. `WithUpdateSafeTx` should afford the following signature:
`jobsDB.WithUpdateSafeTx(brt, func(tx jobsdb.UpdateSafeTx) error) error`

Reply:
which interface, `jobsdb.UpdateSafeTx` or `jobsdb.JobsDB`?